src/charm.py

```diff
@@ -289,6 +297,11 @@ def _on_ingress_revoked(self, event: IngressPerUnitRevokedForUnitEvent):
     def _on_k8s_patch_failed(self, event: K8sResourcePatchFailedEvent):
         self.unit.status = BlockedStatus(event.message)
 
+    def _on_remote_write_endpoints_changed(self, event: RelationChangedEvent) -> None:
+        """Event handler for the remote write endpoint changed event."""
+        logger.debug("Remote write endpoints were changed")
```

Comment:
To my understanding Prometheus should work as usual + send to requested endpoints

@simskij, can you confirm?

Reply:
if it can work as normal - excellent. I was under the impression that enabling the remote write sender automatically put it into agent mode, but @beliaev-maksim's tests seem to suggest otherwise.

Reply:
it goes to the agent mode, but it just replicates the data, like normal mode + push to the endpoint.

_Sure, but the amazing part is that, even with Remote Write, Prometheus still uses a pull model to gather metrics from applications, which gives us an understanding of those different failure modes. After that, we batch samples and series and export, replicate (push) data to the Remote Write endpoints, limiting the number of monitoring unknowns that the central point has!_

source: https://prometheus.io/blog/2021/11/16/agent/